tasks/request_status_for_granule/test/unit_tests/test_request_status_for_granule_unit.py

```diff
@@ -84,3 +92,75 @@ def test_task_granule_id_cannot_be_none(self):
+        job_id = uuid.uuid4().__str__()
+
+        db_connect_info = Mock()
+
+        job_entry = {
+            request_status_for_granule.OUTPUT_JOB_ID_KEY: job_id,
+            request_status_for_granule.OUTPUT_COMPLETION_TIME_KEY: '1999-01-31 09:26:56.66 +02:00'  # todo: utc?
+        }
+        mock_get_job_entry_for_granule.return_value = copy.deepcopy(job_entry)
+
+        file_entries = [{
+            request_status_for_granule.OUTPUT_ERROR_MESSAGE_KEY: "Something went boom."
+        }]
+        mock_get_file_entries_for_granule_in_job.return_value = copy.deepcopy(file_entries)
+
+        result = request_status_for_granule.task(granule_id, db_connect_info, job_id)
+
+        mock_get_file_entries_for_granule_in_job.assert_called_once_with(granule_id, job_id, db_connect_info)
+        mock_get_job_entry_for_granule.assert_called_once_with(granule_id, job_id, db_connect_info)
+
+        expected = job_entry
+        expected[request_status_for_granule.OUTPUT_FILES_KEY] = file_entries
+        self.assertEqual(expected, result)
+
+    @patch('request_status_for_granule.get_most_recent_job_id_for_granule')
+    @patch('request_status_for_granule.get_job_entry_for_granule')
+    @patch('request_status_for_granule.get_file_entries_for_granule_in_job')
+    def test_task_no_job_id_gets(self,
+                                 mock_get_file_entries_for_granule_in_job: MagicMock,
+                                 mock_get_job_entry_for_granule: MagicMock,
+                                 mock_get_most_recent_job_id_for_granule: MagicMock):
+        """
+        If job_id is not given, then it should take a separate request to get it.
+        """
+        granule_id = uuid.uuid4().__str__()
+        job_id = uuid.uuid4().__str__()
+
+        db_connect_info = Mock()
+
+        mock_get_most_recent_job_id_for_granule.return_value = job_id
+
+        job_entry = {
+            request_status_for_granule.OUTPUT_JOB_ID_KEY: job_id,
+            request_status_for_granule.OUTPUT_COMPLETION_TIME_KEY: None
+        }
+        mock_get_job_entry_for_granule.return_value = copy.deepcopy(job_entry)
+
+        file_entries = [{
+            request_status_for_granule.OUTPUT_ERROR_MESSAGE_KEY: None
+        }]
+        mock_get_file_entries_for_granule_in_job.return_value = copy.deepcopy(file_entries)
+
+        result = request_status_for_granule.task(granule_id, db_connect_info, None)
+
+        mock_get_most_recent_job_id_for_granule.assert_called_once_with(granule_id, db_connect_info)
+        mock_get_file_entries_for_granule_in_job.assert_called_once_with(granule_id, job_id, db_connect_info)
+        mock_get_job_entry_for_granule.assert_called_once_with(granule_id, job_id, db_connect_info)
+
+        expected = job_entry
+        del expected[request_status_for_granule.OUTPUT_COMPLETION_TIME_KEY]
+        expected[request_status_for_granule.OUTPUT_FILES_KEY] = [{}, ]
+        self.assertEqual(expected, result)
```

Comment:
Possibly add test to validate schema output for a valid job_id/granule_id combinations.

Notice we test input ok but not necessarily output unless I am missing something.

Reply:
The expected output is constructed on lines 163-165. The test is written to only care about the function under test, so I hesitate to build out a full schema with every property.